aws/table_aws_kinesis_stream.go

```diff
@@ -0,0 +1,305 @@
+	logger := plugin.Logger(ctx)
+	logger.Trace("describeStream")
+
+	// TODO put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	var streamName string
+	if h.Item != nil {
+		streamName = *h.Item.(*kinesis.DescribeStreamOutput).StreamDescription.StreamName
+	} else {
+		quals := d.KeyColumnQuals
+		streamName = quals["stream_name"].GetStringValue()
+	}
+
+	// get service
+	svc, err := KinesisService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &kinesis.DescribeStreamInput{
+		StreamName: aws.String(streamName),
+	}
+
+	// Get call
+	data, err := svc.DescribeStream(params)
+	if err != nil {
+		logger.Debug("describeStream__", "ERROR", err)
+		return nil, err
+	}
+	return data, nil
+}
+
+// API call for Stream Summary
+func describeStreamSummary(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("describeStreamSummary")
+
+	// TODO put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	var streamName string
+	streamName = *h.Item.(*kinesis.DescribeStreamOutput).StreamDescription.StreamName
+
+	// get service
+	svc, err := KinesisService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &kinesis.DescribeStreamSummaryInput{
+		StreamName: aws.String(streamName),
+	}
+
+	// Get call
+	data, err := svc.DescribeStreamSummary(params)
+	if err != nil {
+		logger.Debug("describeStreamSummary__", "ERROR", err)
+		return nil, err
+	}
+	return data, nil
+}
+
+// API call for fetching tag list
+func getAwsKinesisStreamTags(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("getAwsKinesisStreamTags")
+
+	// TODO put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	streamName := *h.Item.(*kinesis.DescribeStreamOutput).StreamDescription.StreamName
+
+	// Create Session
+	svc, err := KinesisService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &kinesis.ListTagsForStreamInput{
+		StreamName: &streamName,
+	}
+
+	// Get call
+	op, err := svc.ListTagsForStream(params)
+	if err != nil {
+		logger.Debug("getAwsKinesisStreamTags", "ERROR", err)
+		return nil, err
+	}
+
+	return op, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func kinesisTagListToTurbotTags(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("ssmTagListToTurbotTags")
+	tagList := d.Value.([]*kinesis.Tag)
+
```

Comment:
Return nil, if `tagList` is null
```suggestion

if tagList == nil {
  return nil, nil
}
```